packages/2018-neighborhood-development/src/state/explore-urban-campsite-sweeps/selectors.js

```diff
@@ -20,3 +22,51 @@ export const catchCampsiteSweepsErrors = createSelector(
   getCampsiteSweepsRequest,
   ({ error }) => error || error,
 );
+
+export const getCampsiteSweepsTimer = createSelector(
+  getCampsiteSweepsRequest,
+  ({ timer }) => timer,
+);
+
+const getCampsiteSweepsDataTimepoints = data => data.map(item => new Date(item.properties.reportdate));
+
+const splitTimepointsToMonths = (timepoints) => {
+  const startMonth = min(timepoints).getMonth();
+  const startYear = min(timepoints).getFullYear();
+  const endMonth = max(timepoints).getMonth();
+  const endYear = max(timepoints).getFullYear();
+  const startDate = new Date(startYear, startMonth);
+  const endDate = new Date(endYear, endMonth, 0);
+  const months = (endDate.getMonth() - startDate.getMonth()) + 1 + ((endDate.getFullYear() - startDate.getFullYear()) * 12);
+  const sections = [...Array(months + 1).keys()];
+  return sections.map(item => new Date(startDate.getFullYear(), startDate.getMonth() + item));
+};
```

Comment:
Oof. Date stuff in vanilla js is so hairy. It might be time to pull in a date lib. I'm thinking maybe [date-fns](https://date-fns.org/) instead of the goto-pick of Moment.

Reply:
Was thinking of creating a new issue for bringing in date-fns rather than addressing this right now - I'd like to clean up some of the other date code and add it to the documentation while I'm at it.